[WIP] sci-geosciences/gpsbabel: 1.6.0 version bump, EAPI 7

cjmayo opened the pull request:
Closes: https://bugs.gentoo.org/689586
Package-Manager: Portage-2.3.69, Repoman-2.3.16
Signed-off-by: Chris Mayo <aklhfex@gmail.com>

---

A fair amount of change shown below. Highlights:
- Get source from GitHub
- zlib[minizip] wasn't being used, and now checks removed upstream from configure
- Update three of the patches
- eautoreconf no longer needed
- upstream renamed configure.in to configure.ac
- Install gui/gmapbase.html, which unfortunately needs another patch to use


```diff
--- gpsbabel-1.5.4-r1.ebuild
+++ gpsbabel-1.6.0.ebuild
@@ -1,26 +1,29 @@
-# Copyright 1999-2017 Gentoo Foundation
+# Copyright 1999-2019 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
+EAPI=7
 
-inherit autotools eutils qmake-utils
+inherit desktop qmake-utils
+
+MY_PV="${PV//./_}"
+MY_P="${PN}_${MY_PV}"
 
 DESCRIPTION="GPS waypoints, tracks and routes converter"
 HOMEPAGE="https://www.gpsbabel.org/ https://github.com/gpsbabel/gpsbabel"
 SRC_URI="
-	https://dev.gentoo.org/~asturm/${P}.tar.gz
+	https://github.com/${PN}/${PN}/archive/${MY_P}.tar.gz
 	doc? ( https://www.gpsbabel.org/style3.css -> gpsbabel.org-style3.css )"
 LICENSE="GPL-2"
 
 SLOT="0"
-KEYWORDS="amd64 ~ppc x86 ~x86-fbsd"
+KEYWORDS="~amd64 ~ppc ~x86 ~x86-fbsd"
 IUSE="doc +gui"
 
 RDEPEND="
 	dev-libs/expat
 	dev-qt/qtcore:5
 	sci-libs/shapelib:=
-	sys-libs/zlib[minizip]
+	sys-libs/zlib
 	virtual/libusb:0
 	gui? (
 		dev-qt/qtgui:5
@@ -31,7 +34,7 @@
 		dev-qt/qtxml:5
 	)
 "
-DEPEND="${RDEPEND}
+BDEPEND="
 	doc? (
 		app-text/docbook-xml-dtd:4.1.2
 		dev-lang/perl
@@ -39,21 +42,23 @@
 	)
 	gui? ( dev-qt/linguist-tools:5 )
 "
+DEPEND="${RDEPEND}"
 
 DOCS=( AUTHORS README.{contrib,igc,mapconverter,md,xmapwpt} )
 
 PATCHES=(
-	"${FILESDIR}"/${PN}-1.4.3-fix_binary_name.patch
-	"${FILESDIR}"/${P}-disable_statistic_uploading.patch
-	"${FILESDIR}"/${P}-disable_update_check.patch
-	"${FILESDIR}"/${P}-disable_version_check.patch
 	"${FILESDIR}"/${P}-use_system_shapelib.patch
 	"${FILESDIR}"/${P}-use_system_zlib.patch
-	"${FILESDIR}"/${P}-xmldoc.patch
-	"${FILESDIR}"/${P}-qt-5.9-{1,2}.patch
+	"${FILESDIR}"/${PN}-1.5.4-xmldoc.patch
+	"${FILESDIR}"/${PN}-1.5.4-disable_statistic_uploading.patch
+	"${FILESDIR}"/${P}-disable_update_check.patch
+	"${FILESDIR}"/${PN}-1.5.4-disable_version_check.patch
+	"${FILESDIR}"/${P}-gmapbase_path.patch
 )
 
 RESTRICT="test" # bug 421699
+
+S="${WORKDIR}/${PN}-${MY_P}"
 
 src_prepare() {
 	default
@@ -62,12 +67,7 @@
 	rm -r shapelib zlib || die
 	sed -i -e "s: zlib\/[a-z]*\.h::g" Makefile.in || die
 
-	mv configure.in configure.ac || die
-	sed -i -e "/^configure:/s/configure.in/configure.ac/" Makefile.in || die
-
 	use doc && cp "${DISTDIR}/gpsbabel.org-style3.css" "${S}"
-
-	eautoreconf
 }
 
 src_configure() {
@@ -107,8 +107,11 @@
 
 	if use gui; then
 		dobin gui/objects/gpsbabelfe
+		insinto /usr/share/${PN}/
+		doins gui/gmapbase.html
 		insinto /usr/share/${PN}/translations/
 		doins gui/gpsbabel*_*.qm
+		doins gui/coretool/gpsbabel*_*.qm
 		newicon gui/images/appicon.png ${PN}.png
 		make_desktop_entry gpsbabelfe ${PN} ${PN} "Science;Geoscience"
 	fi
```

cjmayo commented:
Please don't merge this yet. Translations don't work here, or in 1.5.4-r1, plus I don't really like my new patch. I think I can solve both in one patch and even have a go at upstreaming it.

cjmayo commented:
OK. Updated with a new patch that fixes translations and gmapbase.html. gui makefile generation command changes to `eqmake5 PREFIX=/usr`.

tsteven4 commented:
I don't think this is going to quite work.  If I am interpreting your ebuild file correctly you are copying the gpsbabel supplied translations to /usr/share/gpsbabel/translations, and patching mainwindow.cc to find these.  But we will also look for the Qt supplied translations there (qt_\*.qm), and they won't be there.
The result will be missing translations, look at the Help and Close button at the bottom:
LANG=de_DE.UTF-8 ./gpsbabelfe
![image](https://user-images.githubusercontent.com/13596209/62553585-d26de500-b82c-11e9-9f31-fd18c8a01e98.png)
They should look like
![image](https://user-images.githubusercontent.com/13596209/62553682-06490a80-b82d-11e9-886f-3beb1114e3d6.png)
The translation for these is provided by Qt.
Note that Qt5 uses translation meta catalogs.  If you use those you need another patch to gpsbabel, and you need to included the appropriate module translation files.  On the other hand, the gpsbabel package target will concatenate only the required module translation files into one qt_\*.qm file as recommended by Qt, then you just need it.

tsteven4 commented:
You may not need you zlib patch at all.  Because you are using the configure option --with-zlib=system -lz should be added to @LIBS@ by configure, @ZLIB@ should be replaced by nothing in the generate Makefile, and the Makfile variable ZLIB will be unreferenced.

a17r commented on sci-geosciences/gpsbabel/gpsbabel-1.6.0.ebuild:
```diff
@@ -0,0 +1,118 @@
+SRC_URI="
+	https://github.com/${PN}/${PN}/archive/${MY_P}.tar.gz
+	doc? ( https://www.gpsbabel.org/style3.css -> gpsbabel.org-style3.css )"
+LICENSE="GPL-2"
```
per skel.ebuild LICENSE should be one line below, in the same block as `SLOT`

cjmayo replied:
Done in the new commit, and moved RESTRICT to the end of the same block.

a17r commented on sci-geosciences/gpsbabel/gpsbabel-1.6.0.ebuild:
```diff
@@ -0,0 +1,118 @@
+LICENSE="GPL-2"
+
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86 ~x86-fbsd"
```
`x86-fbsd` is dead, please drop ;)

cjmayo replied:
Dropped in the new commit.

cjmayo commented:
This one got a bit complicated. I'd already moved on from the ebuild here while working with @tsteven4 upstream. I've added my latest ebuild as an additional commit here (the original commit is unchanged, just rebased on current master) to show the changes.
Alas most of these patches aren't accepted, the shapelib change was effectively vetoed and the GUI solution is still open.
I feel a bit stuck here. I don't want to spend time going back to the original ebuild but the new one has quite a lot for Gentoo to carry on its own.

cjmayo commented:
> You may not need you zlib patch at all. Because you are using the configure option --with-zlib=system -lz should be added to @libs@ by configure, @zlib@ should be replaced by nothing in the generate Makefile, and the Makfile variable ZLIB will be unreferenced.

zlib patch dropped in new commit.

cjmayo commented:
> I don't think this is going to quite work. If I am interpreting your ebuild file correctly you are copying the gpsbabel supplied translations to /usr/share/gpsbabel/translations, and patching mainwindow.cc to find these. But we will also look for the Qt supplied translations there (qt_*.qm), and they won't be there.
> The result will be missing translations, look at the Help and Close button at the bottom:
> LANG=de_DE.UTF-8 ./gpsbabelfe

Can't remember if we had fixed that in the patches or not. I may well not have German Qt5 translations installed either.

cjmayo commented:
Locating the Qt translations as well is fixed in the patches used. I do now have those translations installed (dev-qt/qttranslations) and have checked.

cjmayo commented:
Perhaps the best way to get something in now is not to install the GUI: https://github.com/gentoo/gentoo/pull/14242.

akhuettel commented:
Chris, first of all, sorry I wasnt aware of the PRs and started the work myself. But maybe we can combine things somehow. Which one of the two PRs is "current"? (Let's close the other one then.)

cjmayo commented:
I would go with https://github.com/gentoo/gentoo/pull/14242 for now. There's no upstream solution on the GUI yet - I'll keep up-to-date with it though.
